app/uk/gov/hmrc/servicedependencies/service/DependencyService.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.servicedependencies.service
+
+import com.google.inject.Inject
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.servicedependencies.connector.TeamsAndRepositoriesConnector
+import uk.gov.hmrc.servicedependencies.model.RepoType.Other
+import uk.gov.hmrc.servicedependencies.model.{DependencyScope, MetaArtefact, MetaArtefactDependency}
+import uk.gov.hmrc.servicedependencies.persistence.MetaArtefactRepository
+import uk.gov.hmrc.servicedependencies.persistence.derived.DerivedDependencyRepository
+import uk.gov.hmrc.servicedependencies.util.DependencyGraphParser
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DependencyService @Inject()(
+                                   metaArtefactRepository: MetaArtefactRepository,
+                                   derivedDependencyRepository: DerivedDependencyRepository,
+                                   teamsAndRepositoriesConnector : TeamsAndRepositoriesConnector
+                                 )(implicit ec: ExecutionContext, hc: HeaderCarrier) {
+
+  private def artefactVersionControl(metaArtefact: MetaArtefact): Future[Option[MetaArtefact]] = {
+    metaArtefactRepository.find(metaArtefact.name).map {
+      case Some(value) =>
+        if (metaArtefact.version.compare(value.version) > 0) {
+          Some(metaArtefact)
+        } else {
+          None
+        }
+      case None => Some(metaArtefact)
+    }
+  }
+  def setArtefactDependencies(metaArtefact: MetaArtefact): Future[Unit] = {
+    artefactVersionControl(metaArtefact).flatMap {
```

Comment:
`artefactVersionControl` is hard to understand without reading the indirection - could replace with `def isLatest(metaArtefact: MetaArtefact): Future[Boolean]` and use `Some(metaArtefact).filterA(isLatest)`